### DISCUSSION TITLE
Holes in Skull

### DISCUSSION THREAD

**Discussion by uckhill** [simnibs/simnibs#163] [7.Creating head models]
Hello all,

I am new to using SIMNIBS but have some experience with FreeSurfer. I am using the Windows GUI to create head models from MRI scans. I have recently processed one subject, using both T1 and T2 scans, and found some holes in the skull. Are these gaps in the skull actually present, or the result of a faulty segmentation? Thanks in advance. I look forward to seeing what all this program can do. 
![rightskullmesh](https://github.com/simnibs/simnibs/assets/138915870/81d5fe07-575c-44e9-814f-096674fd8669)

**Comment by oulap** [comment]
That looks a bit strange. There's a file called tissue_labeling_upsampled.nii.gz, along with the upsampled versions of the T1 and T2-weighted scans, inside the m2m-folder in sub-folder called label_prep. Can you open that in freeview (there's also a LUT-file you can use in freeview) and check if it looks okay?

It also seems that the field-of-view is cropped so that there's no neck in the scan. If that is indeed the case please pass the --noneck flag to the charm command.

**Reply by KTZ228** [reply]
I don't know if this issue was resolved privately but if it was I would be interested in the solution. 
We have been correcting for the holes in the skull manually by extending it's boundaries.

I checked the `tissue_labeling_upsampled` file and it does not seem to contain the holes found in the original segmented image.
It is important to note however, that both of the segmented images underestimate the thickness of the skull (specifically in the temporal window).

**Reply by oulap** [reply]
I haven't heard back from OP. In any case the tissue_labeling_upsampled is the file from which the mesh is generated from, so if that does not contain holes the mesh shouldn't either. The file is post-processed however, specifically we enforce a layer of CSF around the brain, which might lead to the underestimation around the temporal regions.

**Reply by KTZ228** [reply]
Aah I see, that makes sense.
After the segmented image is created we stop using SimNIBS and assign acoustic properties to the layer labels. 
Since skull tissue is bound to have a much larger effect on our results than CSF, we would be interested in a model without an enforced CSF layer. Is this a setting that can be enabled?

**Reply by oulap** [reply]
charm has a --debug flag, which writes out a few more files in the label_prep folder including a labeling before any morphological operations have been applied. You can try running with the --debug flag and checking if the additional files look better in the temporal region. They'll most likely be noisier over all as the morphological operations are designed to clean up small segmentation error.

**Reply by KTZ228** [reply]
Thank you for your suggestions, the `before_morpho` file indeed seems to more accurately depict the thickness of the skull tissue, particularly in the temporal window. However, missing the morphological smoothing processes is not ideal.
I have seen multiple researchers in the transcranial ultrasound field struggle with this so it would benefit everyone involved if we can more accurately estimate the effects the skull has on the sonication of grey and white matter.

So my question to you is this: would it be possible to add a new argument to the charm function to turn off the expansion of the CSF borders in a future release?
